Horan_Colby/Assignments/library.js

```diff
@@ -0,0 +1,37 @@
+var _ = {
+    map: function(arr, callback) {
+        var newarr = []
+        for(var i = 0;i < arr.length;i++){
+            newarr.push(callback(arr[i]))
+        }
+    },
+    reduce: function(arr, callback){
+        x = 0
+        for(var i = 0;i <arr.length;i++){
+            x += callback(arr[i])
+        }
+    },
+    find: function(arr, callback) {   
+        for(var i = 0;i < arr.length;i++){
+            if (arr[i] == callback){
+                return arr[i]
+            }
+        }
+    },
+    filter: function(arr, callback) { 
+        newarr = []
+        for (var i = 0;i < arr.length;i++){
+            if (arr[i] == callback){
+                newarr.push(callback(arr[i]))
+            }
+        }
+    },
+    reject: function() { 
+        newarr = []
+        for (var i = 0;i < arr.length;i++){
+            if (arr[i] != callback){
```

Comment:
Make sure to invoke these!  How do you know they work?  I see some potential issues with what these functions return.